src/libraries/Microsoft.Extensions.Hosting/src/HostOptions.cs

```diff
@@ -10,6 +11,16 @@ namespace Microsoft.Extensions.Hosting
     /// </summary>
     public class HostOptions
     {
+        public HostOptions()
+        {
+            var timeoutSeconds = Environment.GetEnvironmentVariable("DOTNET_SHUTDOWNTIMEOUTSECONDS");
```

Comment:
I guess I expected the fix here was to inject an `IConfiguration` into the HostOptions class - just like the WebHost does. Is there a reason we aren't taking that approach?

https://github.com/dotnet/aspnetcore/blob/fd1f1c355984f76070d8e32c891e6bac0bd96fdf/src/Hosting/Hosting/src/Internal/WebHostOptions.cs#L19-L40

Reply:
Yes, I also think we shouldn't need to use GetEnvironmentVariable here. then the testing can be done using IConfiguration as well

Reply:
@eerhardt Options must have parameterless constructor, so `HostOptions` will have two constructors (Changes to this file must follow the https://aka.ms/api-review process). But we can not just get `HostOptions` with `IConfiguration` from DI when we have parameterless constructor, can we?

Reply:
DI will pick the constructor with the longest signature. So while we will keep the parameter less ctor for anyone creating this class manually, DI will always create it using the IConfiguration.

Reply:
@eerhardt afaik this rule does not work for IOptions injection https://github.com/dotnet/runtime/blob/0eb6c4eba8b391db117a22bf60d9eaec9695fa97/src/libraries/Microsoft.Extensions.Options/src/IOptions.cs#L12-L19
https://github.com/dotnet/runtime/blob/0eb6c4eba8b391db117a22bf60d9eaec9695fa97/src/libraries/Microsoft.Extensions.Options/src/Options.cs#L11-L14

Reply:
Ah, you are right - I was missing that it is the `OptionsMonitor` that gets injected in DI, and not the Options object itself. Looking at ASP.NET, they don't use DI at all to create the WebHostOptions object - they do it themselves.

One way to solve this issue is to use a `.Configure<HostOptions>` call to hook the configuration up to the options. Here's an example:

https://github.com/eerhardt/runtime/commit/01ce98195405b51080e5a80e53902abe1f3fbe0b

Reply:
Updated